script/helpers/PantosHubRedeployer.s.sol

```diff
@@ -120,33 +116,117 @@ abstract contract PantosHubRedeployer is PantosHubDeployer {
                             blockchainId,
                             externalTokenRecord.externalToken
                         );
-                        console2.log(
-                            "New PantosHub.registerExternalToken(%s, %s, %s)",
-                            address(_ownedTokens[i]),
-                            blockchain.name,
-                            externalTokenRecord.externalToken
-                        );
                     }
                 }
             }
         }
+        // unregister tokens from old hub after all registered at new hub
+        unregisterTokensFromOldHub();
+    }
+
+    function registerTokenAtNewHub(
+        IPantosHub newPantosHubProxy,
+        PantosToken token,
+        uint256 stake
+    ) public onlyPantosHubRedeployerInitialized {
+        PantosTypes.TokenRecord memory tokenRecord = newPantosHubProxy
+            .getTokenRecord(address(token));
+        if (!tokenRecord.active) {
+            newPantosHubProxy.registerToken(address(token), stake);
+            console2.log("New PantosHub.registerToken(%s)", address(token));
+        } else {
+            console2.log(
+                "Token already registered; skipping registerToken(%s)",
+                address(token)
+            );
+        }
+    }
+
+    function registerExternalTokenAtNewHub(
+        IPantosHub newPantosHubProxy,
+        PantosToken token,
+        uint256 blockchainId,
+        string memory externalToken
+    ) public onlyPantosHubRedeployerInitialized {
+        PantosTypes.ExternalTokenRecord
+            memory externalTokenRecord = newPantosHubProxy
+                .getExternalTokenRecord(address(token), blockchainId);
+
+        if (!externalTokenRecord.active) {
+            newPantosHubProxy.registerExternalToken(
+                address(token),
+                blockchainId,
+                externalToken
+            );
+            console2.log(
+                "New PantosHub.registerExternalToken(%s, %d, %s)",
+                address(token),
+                blockchainId,
+                externalToken
+            );
+        } else {
+            console2.log(
+                "External token already registered;"
+                "skipping PantosHub.registerExternalToken(%s, %d, %s)",
+                address(token),
+                blockchainId,
+                externalToken
+            );
+        }
+    }
+
+    function approvePanTokensForNewHub(
+        IPantosHub newPantosHubProxy
+    ) public onlyPantosHubRedeployerInitialized {
+        uint256 minimumTokenStake = newPantosHubProxy.getMinimumTokenStake();
+        uint256 panTokensNeeded = minimumTokenStake * _ownedTokens.length;
+
+        uint256 panTokensAllowance = _pantosToken.allowance(
+            msg.sender,
+            address(newPantosHubProxy)
+        );
+
+        if (panTokensAllowance < panTokensNeeded) {
+            _pantosToken.approve(address(newPantosHubProxy), 0);
```

Comment:
Why approving 0 is necessary here?
As far as I know, the approval on the next line would just overwrite any older approval

Reply:
In case not sure: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/c3f8b760ad9d0431f33e8303658ccbdcc1610f24/contracts/token/ERC20/ERC20.sol#L280

The 0 gets overwritten

Reply:
I followed this recommendation:
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v5.0.2/contracts/token/ERC20/IERC20.sol#L58  

Since its a one off script, Its not much overhead.

Reply:
Oh, I see. Thanks for pointing it out. I would keep it like this then